test/e2e/subscription_e2e_test.go

```diff
@@ -2534,6 +2534,147 @@ var _ = Describe("Subscription", func() {
+				addBundleUnpackTimeoutOGAnnotation(context.Background(), ctx.Ctx().Client(), ogNN, "5m")
+
+				By("updating the catalog with a fixed v0.2.0 bundle image marked deprecated at all levels")
+				provider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, subscriptionTestDataBaseDir, "example-operator.v0.2.0-deprecations.yaml"))
+				Expect(err).To(BeNil())
+				err = magicCatalog.UpdateCatalog(context.Background(), provider)
+				Expect(err).To(BeNil())
+
+				By("waiting for the subscription to have v0.2.0 installed")
+				sub, err := fetchSubscription(crc, generatedNamespace.GetName(), subName, subscriptionHasCurrentCSV("example-operator.v0.2.0"))
+				Expect(err).Should(BeNil())
+
+				By("checking for the bundle deprecated condition")
+				bundleCondition := sub.Status.GetCondition(operatorsv1alpha1.SubscriptionBundleDeprecated)
+				Expect(bundleCondition.Status).To(Equal(corev1.ConditionTrue))
+				bundleDeprecatedMessage := bundleCondition.Message
+
+				By("updating the catalog with a fixed v0.3.0 bundle marked partially deprecated")
+				provider, err = NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, subscriptionTestDataBaseDir, "example-operator.v0.3.0-deprecations.yaml"))
+				Expect(err).To(BeNil())
+				err = magicCatalog.UpdateCatalog(context.Background(), provider)
+				Expect(err).To(BeNil())
+
+				By("waiting for the subscription to have v0.3.0 installed")
+				sub, err = fetchSubscription(crc, generatedNamespace.GetName(), subName, subscriptionHasCurrentCSV("example-operator.v0.3.0"))
+				Expect(err).Should(BeNil())
+
+				By("checking for the deprecated conditions")
+				// Operator is deprecated at only Package and Channel levels
+				packageCondition := sub.Status.GetCondition(operatorsv1alpha1.SubscriptionPackageDeprecated)
+				Expect(packageCondition.Status).To(Equal(corev1.ConditionTrue))
+				channelCondition := sub.Status.GetCondition(operatorsv1alpha1.SubscriptionChannelDeprecated)
+				Expect(channelCondition.Status).To(Equal(corev1.ConditionTrue))
+				bundleCondition = sub.Status.GetCondition(operatorsv1alpha1.SubscriptionBundleDeprecated)
+				Expect(bundleCondition.Status).To(Equal(corev1.ConditionUnknown))
+
+				By("verifying that a roll-up condition is present not containing bundle deprecation condition")
+				// Roll-up condition should be present and contain deprecation messages from Package and Channel levels
+				rollUpCondition := sub.Status.GetCondition(operatorsv1alpha1.SubscriptionDeprecated)
+				Expect(rollUpCondition.Status).To(Equal(corev1.ConditionTrue))
+				Expect(rollUpCondition.Message).To(ContainSubstring(packageCondition.Message))
+				Expect(rollUpCondition.Message).To(ContainSubstring(channelCondition.Message))
+				Expect(rollUpCondition.Message).ToNot(ContainSubstring(bundleDeprecatedMessage))
+			})
+
+			It("should report deprecated status when catalog is updated to deprecate an installed bundle", func() {
```

Comment:
This test case differs in that it runs a catalog update which does not result in an operator upgrade. I left it because before we changed our strategy to update status in the Subscription reconciler this test case would fail because the cache entries of installed bundles did not contain deprecation info.

Reply:
Ah, gotcha. Thanks for the additional context!